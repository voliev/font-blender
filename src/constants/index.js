// *** ACTION TYPES ***

// Background
export const SET_BACKGROUND_COLOR = "SET_BACKGROUND_COLOR";

// Filteres
export const SET_CATEGORIES_FILTER = "SET_CATEGORIES_FILTER";
export const SET_SORTING_ORDER = "SET_SORTING_ORDER";
export const SET_SORTING_PARAM = "SET_SORTING_PARAM";
export const SET_TITLE_FILTER = "SET_TITLE_FILTER";

// Fonts
export const FETCH_FONTS_DATA = "FETCH_FONTS_DATA";
export const FETCH_FONTS_DATA_SUCCESS = "FETCH_FONTS_DATA_SUCCESS";
export const FETCH_FONTS_DATA_FAILURE = "FETCH_FONTS_DATA_FAILURE";
export const SET_FONTS = "SET_FONTS";
export const UPDATE_FONT = "UPDATE_FONT";

// Notifications
export const ADD_NOTIFICATION = "ADD_NOTIFICATION";
export const REMOVE_NOTIFICATION = "REMOVE_NOTIFICATION";

// Showcase
export const DISPLAY_SHOWCASE_TEXT = "DISPLAY_SHOWCASE_TEXT";
export const DISPLAY_STYLES = "DISPLAY_STYLES";
export const DISPLAY_SHOWCASE_FONT_PREVIEW = "DISPLAY_SHOWCASE_FONT_PREVIEW";

// Styles
export const SET_DEFAULT_STATE = "SET_DEFAULT_STATE";
export const UNDO_RESET = "UNDO_RESET";

// Text styles
export const SET_STYLES_INHERITANCE = "SET_STYLES_INHERITANCE";
export const UPDATE_ELEMENT_STYLES = "UPDATE_ELEMENT_STYLES";

// Selected fonts
export const ADD_SELECTED_FONT = "ADD_SELECTED_FONT";
export const CLEAR_SELECTED_FONTS = "CLEAR_SELECTED_FONTS";
export const REMOVE_FONT_FROM_SELECTED = "REMOVE_FONT_FROM_SELECTED";
export const UNDO_DESELECTED = "UNDO_DESELECTED";

// Type scale
export const SET_DEFAULT_TYPE_SCALE_STATE = "SET_DEFAULT_TYPE_SCALE_STATE";
export const SET_FONT_SIZE = "SET_FONT_SIZE";
export const SET_TYPE_SCALE = "SET_TYPE_SCALE";

// *** MISCELLANEOUS ***

// Google Fonts API endpoint URL
export const GOOGLE_FONTS_API_ENDPOINT =
  "https://www.googleapis.com/webfonts/v1/webfonts";
// Fonts' bundle size that should be prefetched
export const BUNDLE_SIZE = 50;
// Available text elements for styling
export const ELEMENTS = ["body", "h1", "h2", "h3", "h4", "li", "small"];
// Font's character range that needs to be prefetched
export const CHAR_RANGE =
  "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
// Font showcase string
export const SHOWCASE_STRING =
  "That's the beauty of music. They can't get that from you...";
// Default showcase font size
export const DEFAULT_FONT_SIZE = 16;
// Available type scale values
export const SCALE_VALUES = {
  1.067: "Minor Second",
  1.125: "Major Second",
  1.2: "Minor Third",
  1.25: "Major Third",
  1.333: "Perfect Fourth",
  1.414: "Augmented Fourth",
  1.5: "Perfect Fifth",
  1.618: "Golden Ratio"
};
// Minimum gap between the color picker icon and viewport
// edge required to place the color picker
export const MIN_GAP = 300;
// Notification snackbar visibility timeout
export const TIMEOUT = 10000;
