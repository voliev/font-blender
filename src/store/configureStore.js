import { createStore, combineReducers, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import fonts from "../reducers/fonts";
import filters from "../reducers/filters";
import selectedFonts from "../reducers/selectedFonts";
import showcase from "../reducers/showcase";
import styles from "../reducers/styles";
import notifications from "../reducers/notifications";

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default () => {
  const store = createStore(
    combineReducers({
      fonts,
      filters,
      selectedFonts,
      showcase,
      styles,
      notifications
    }),
    composeEnhancers(applyMiddleware(thunk))
  );

  return store;
};
