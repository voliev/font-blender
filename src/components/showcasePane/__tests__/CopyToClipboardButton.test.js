import React from "react";
import { shallow } from "enzyme";
import configureStore from "redux-mock-store";
import Clipboard from "clipboard";
import Copy from "@material-ui/icons/FileCopyOutlined";
import CopyToClipboardButtonConnected, {
  CopyToClipboardButton
} from "../CopyToClipboardButton";
import { ADD_NOTIFICATION } from "../../../constants";
import state from "../../../fixtures/state";

const setup = (renderFn = shallow, propsOverrides = {}, connected = false) => {
  const props = {
    addNotification: jest.fn(),
    notificationActive: false,
    target: "code-snippet",
    ...propsOverrides
  };

  if (connected) {
    const mockStore = configureStore();
    const store = mockStore(state);

    return {
      store,
      wrapper: renderFn(
        <CopyToClipboardButtonConnected store={store} target={props.target} />
      )
    };
  }

  return {
    ...props,
    wrapper: renderFn(<CopyToClipboardButton {...props} />)
  };
};

afterEach(() => {
  jest.clearAllMocks();
});

it("should render without crashing", () => {
  const { wrapper } = setup();

  expect(wrapper).toMatchSnapshot();
});

it("should set corresponding target", () => {
  const { wrapper } = setup();

  expect(wrapper.find("button").prop("data-clipboard-target")).toBe(
    "#code-snippet"
  );
});

it("should set corresponding icon", () => {
  const { wrapper } = setup();

  expect(wrapper.find(Copy).length).toBe(1);
  expect(wrapper.find(Copy).prop("fontSize")).toBe("small");
});

it("should create new Clipboard instance", () => {
  const { wrapper } = setup();

  expect(wrapper.instance().clipboard).toBeInstanceOf(Clipboard);
});

it("should remove Clipboard object instance on unmount", () => {
  const { wrapper } = setup();
  const destroy = jest.spyOn(Clipboard.prototype, "destroy");

  expect(wrapper.instance().clipboard).toBeInstanceOf(Clipboard);

  wrapper.unmount();

  expect(destroy).toHaveBeenCalledTimes(1);
});

it("should clear selection on clipboard `copy` event", () => {
  const clearSelection = jest.fn();
  const { wrapper } = setup();

  wrapper.instance().handleCopySuccess({ clearSelection });

  expect(clearSelection).toHaveBeenCalledTimes(1);
});

it("should add new notification", () => {
  const { wrapper, addNotification } = setup();

  wrapper.instance().handleCopySuccess({ clearSelection: jest.fn() });

  expect(addNotification).toHaveBeenCalledTimes(1);
  expect(addNotification).toHaveBeenLastCalledWith("Copied to clipboard");

  wrapper.instance().handleCopyError();

  expect(addNotification).toHaveBeenCalledTimes(2);
  expect(addNotification).toHaveBeenLastCalledWith(
    "Unable to copy to clipboard"
  );

  wrapper.setProps({ notificationActive: true });

  wrapper.instance().handleCopySuccess({ clearSelection: jest.fn() });

  expect(addNotification).toHaveBeenCalledTimes(2);

  wrapper.instance().handleCopyError();

  expect(addNotification).toHaveBeenCalledTimes(2);
});

it("should dispatch an action", () => {
  const { wrapper, store } = setup(shallow, undefined, true);

  wrapper.prop("addNotification")("Copied to clipboard");

  const actions = store.getActions();

  expect(actions).toContainEqual({
    type: ADD_NOTIFICATION,
    text: "Copied to clipboard",
    group: "clipboard"
  });
});
