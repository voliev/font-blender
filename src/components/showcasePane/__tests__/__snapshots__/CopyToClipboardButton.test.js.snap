// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<button
  className="CopyToClipboardButton btn-blue-rect"
  data-clipboard-target="#code-snippet"
  disabled={false}
  title="Copy to clipboard"
  value="copy"
>
  <span
    className="icon"
  >
    <FileCopyOutlinedIcon
      fontSize="small"
    />
  </span>
  Copy to clipboard
</button>
`;
